File: src/bin/0435-non-overlapping-intervals.py
# @leetup=info id=435 lang=python3 slug=non-overlapping-intervals



from typing import Optional

# @leetup=code
class Solution:
    def eraseOverlapIntervals(self, intervals: list[list[int]]) -> int:
        intervals.sort()
        res = 0
        end = intervals[0][1]
        for interval in intervals[1:]:
            if interval[0] < end:
                res += 1
                end = min(end, interval[1])
            else:
                end = interval[1]

        return res


# @leetup=code

File: src/bin/0435-non-overlapping-intervals.rs
// @leetup=info id=435 lang=rust slug=non-overlapping-intervals

// @leetup=code
impl Solution {
    pub fn erase_overlap_intervals(mut intervals: Vec<Vec<i32>>) -> i32 {
        intervals.sort();
        let mut res = 0;
        let mut end = intervals[0][1];
        for interval in intervals.into_iter().skip(1) {
            if interval[0] < end {
                res += 1;
                end = end.min(interval[1])
            } else {
                end = interval[1]
            }
        }
        res
    }
}
// @leetup=code

struct Solution;

fn main() {
    let _ = Solution::erase_overlap_intervals;
}
